logging/src/main/java/org/jboss/as/logging/validators/SuffixValidator.java

```diff
@@ -53,10 +53,21 @@ public void validateParameter(String parameterName, ModelNode value) throws Oper
         if (value.isDefined()) {
             final String suffix = value.asString();
             try {
-                if (denySeconds && (suffix.contains("s") || suffix.contains("S"))) {
-                    throw createOperationFailure(LoggingLogger.ROOT_LOGGER.suffixContainsMillis(suffix));
-                }
                 new SimpleDateFormat(suffix);
+                if (denySeconds) {
+                    for (int i = 0; i < suffix.length(); i++) {
+                        char c = suffix.charAt(i);
+                        if (c == '\'') {
+                            c = suffix.charAt(++i);
+                            while (c != '\'') {
+                                c = suffix.charAt(++i);
```

Comment:
Added unit tests. I first validate the suffix on L56 and then check for 's' in order to prevent checking invalid formats like "m'suffix" (one single quote) which in this case will throw StringIndexOutOfBoundsException.